rules-reviewed/jboss-eap/jboss-eap4/eap4-xml-config.windup.xml

```diff
@@ -0,0 +1,344 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ruleset id="eap4-eap6"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns:wu="http://windup.jboss.org/schema/jboss-ruleset"
+         xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+
+    <metadata>
+        <description>
+            This ruleset provides analysis of JBoss EAP 4 configuration XML files that may require individual attention
+            when migrating to JBoss EAP 6+. Mainly deals with datasources, JBoss Web and EJB configuration.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,2.3.0.Final"/>
+            <addon id="org.jboss.windup.rules,windup-rules-java,2.3.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="eap" versionRange="[4,5)"/>
+        <targetTechnology id="eap" versionRange="[6,)"/>
+        <tag>jboss-web</tag>
+        <tag>datasources</tag>
+        <tag>ejb</tag>
+        <tag>reviewed-2015-08-28</tag>
+    </metadata>
+
+
+    <!-- Rules converted from Tubame JBossMigrationKnowhowEAP4toEAP6-En.xml -->
+
+    <rules>
+
+        <rule id="eap4-eap6-25000">
+            <when>
+                <xmlfile matches="//datasources/local-tx-datasource/connection-url[contains(text(),'jdbc:postgres')]" in="{*}-ds.xml"/>
+            </when>
+            <perform>
+                <hint effort="5" severity="MANDATORY">
+                    <message>PostgreSQL JDBC URL. Migrate the datasource settings manually.</message>
```

Comment:
This needs to use complete sentences and provide more information in the message about what is actually going on, and what needs to change. For instance. Should the user delete this file, or this piece of the file, and instead configure the datasource using the Admin UI? Many of these hints are not clear and need more descriptive help.